Homework2: Tatiana Smatova

SmatovaTatiana opened the pull request:
### PR checklist
Please make sure you've passed PR checklist
- [x] my PR passes CI checks (flake8 and pycodestyle).
- [x] my PR named according to guidelines.
- [x] my PR contains complete tasks descriptions for all tasks.
- [x] my PR made from the fork feature branch to appropriate branch.
- [x] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

SmatovaTatiana committed:
com 1

SmatovaTatiana committed:
del

SmatovaTatiana committed:
first commit

SmatovaTatiana committed:
first commit

SmatovaTatiana committed:
task1_first commit

SmatovaTatiana committed:
create folder homework2. Moved task1.py to this folder

SmatovaTatiana committed:
add task7

SmatovaTatiana committed:
1-st line in docstring was too long. Changed it

SmatovaTatiana committed:
1-st commit for task_5

SmatovaTatiana committed:
1-st commit for task_6

SmatovaTatiana committed:
comment was too long

SmatovaTatiana committed:
comment was too long

SmatovaTatiana committed:
1-st commit for task2

SmatovaTatiana committed:
1-st commit for task3

SmatovaTatiana committed:
1-st commit for task3

SmatovaTatiana committed:
correct print template and input line

SmatovaTatiana committed:
the docstring line was too long

SmatovaTatiana committed:
correct print format

SmatovaTatiana committed:
correct print format

SmatovaTatiana committed:
template line was too long

SmatovaTatiana committed:
1-st commit for task 8

SmatovaTatiana committed:
deleted trailing whitespace

SmatovaTatiana committed:
deleted trailing whitespace

belovmd commented on src/homework2/task2.py:
```diff
@@ -0,0 +1,19 @@
+"""
+my_string = input('Input the string: ')
+template = "The longest word is : {longest_word}"
+punctuation_marks = '.,/?!-:;'
```
what about `{)}['"` and others?

belovmd commented on src/homework2/task3.py:
```diff
@@ -0,0 +1,13 @@
+"""Вводится строка. Требуется удалить из нее повторяющиеся символы и все пробелы.
+Например, если было введено "abc cde def", то должно быть выведено "abcdef"."""
+# my_string = 'The Ultimate Question of Life, the Universe, and Everything!..'
```
redundant code in PR

belovmd commented on src/homework2/task3.py:
```diff
@@ -0,0 +1,13 @@
+my_string = my_string.replace(' ', '')
+new_string = my_string[0]
+template = "The string of unique symbols is : {new_string}"
+for i in range(0, len(my_string)):
```
redundant 0

belovmd commented on src/homework2/task4.py:
```diff
@@ -0,0 +1,18 @@
+           "\nNumber of lowercase latin letters : {lower_case}"
+upper_case = 0
+lower_case = 0
+latin_alph_upper = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
```
str.isupper(), str.islower()

belovmd commented on src/homework2/task5.py:
```diff
@@ -0,0 +1,20 @@
+if n < 0:
+    print('Negative')
+elif n <= 1:
+    Fib_n = n
```
wrong naming

belovmd commented on src/homework2/task5.py:
```diff
@@ -0,0 +1,20 @@
+else:
+    Fib_n_2 = 0
+    Fib_n_1 = 1
+    Fib_n = Fib_n_2 + Fib_n_1
```
a, b = a+b, a

belovmd commented on src/homework2/task5.py:
```diff
@@ -0,0 +1,20 @@
+    Fib_n_2 = 0
+    Fib_n_1 = 1
+    Fib_n = Fib_n_2 + Fib_n_1
+    for i in range(0, n - 2):
```
redundant 0

belovmd commented on src/homework2/task6.py:
```diff
@@ -0,0 +1,28 @@
+"""
+x = int(input('Введите число больше 0: '))
+
+# определяем разрядность введенного числа:
```
redundant comments

belovmd commented on src/homework2/task6.py:
```diff
@@ -0,0 +1,28 @@
+
+# если значения равны, перезаписываем исходное число, откидываем при этом крайние цифры,
+# возвращаемся к проверке условия
+    else:
```
too complex solution

belovmd commented on src/homework2/task7.py:
```diff
@@ -0,0 +1,15 @@
+a = float(input('Введите 1-ю сторону треугольника: '))
+b = float(input('Введите 1-ю сторону треугольника: '))
+c = float(input('Введите 1-ю сторону треугольника: '))
+if (a + b) >= c and (b + c) >= a and (a + c) >= b:
```
redundant parentheses

belovmd commented on src/homework2/task8.py:
```diff
@@ -0,0 +1,198 @@
+"""
+my_str_ = input('Input the string: ')
+
+# проверяем длину введенной строки:
```
too many redundant comments

belovmd commented on src/homework2/task8.py:
```diff
@@ -0,0 +1,198 @@
+if my_str_.istitle():
+    print('The capitalized string is :', my_str_)
+
+# преобразуем строку в список слов и создаем пустую строку
```
redundant comments

belovmd commented on src/homework2/task8.py:
```diff
@@ -0,0 +1,198 @@
+    print(template.format(new_line=new_line))
+
+# когда решила задачу, нашла существующий метод строк. Решение практически в одну команду :(
+# print(my_str_.title())
```
what is it?

belovmd commented on src/homework2/task8.py:
```diff
@@ -0,0 +1,198 @@
+    print()
+
+# после долгих изучений свойств функции print, найдено решение из 2-х строк (по условию задачи)
+# тест на сайте не прошел, хотя условия соблюдены и код работает
```
it doesn't work if it doesn't pass tests

belovmd reviewed (changes requested):
you can add one more advanced task for higher score

SmatovaTatiana committed:
add punctuation marks

SmatovaTatiana committed:
add punctuation marks

SmatovaTatiana committed:
add punctuation marks

SmatovaTatiana committed:
correct range in for-loop. Deleted redundant code in PR

SmatovaTatiana committed:
used the recommended str.isupper() and str.islower()

SmatovaTatiana committed:
used the recommended str.isupper() and str.islower()

SmatovaTatiana committed:
changed name and used the recommended a, b = b, a + b

SmatovaTatiana committed:
deleted redundant parentheses

SmatovaTatiana committed:
first version was too complex

SmatovaTatiana committed:
Deleted redundant comments.
Task 8_4: I have 2 solutions. One of them is simple, but contains more than 2 lines. The second solution consists of 2 lines, but does not pass the test. I don't know why. My Output and expected Output are the same.

SmatovaTatiana committed:
Deleted redundant comments.
Task 8_4: I have 2 solutions. One of them is simple, but contains more than 2 lines. The second solution consists of 2 lines, but does not pass the test. I don't know why. My Output and expected Output are the same.

belovmd commented on src/homework2/task2.py:
```diff
@@ -0,0 +1,19 @@
+"""
+my_string = input('Input the string: ')
+template = "The longest word is : {longest_word}"
+punctuation_marks = r'\'"#!$%&()*+,-./:;<=>?@[\]^_{|}~'
```
string.punctuation :)

belovmd commented on src/homework2/task3.py:
```diff
@@ -0,0 +1,12 @@
+my_string = my_string.replace(' ', '')
+new_string = my_string[0]
+template = "The string of unique symbols is : {new_string}"
+for i in range(len(my_string)):
```
for element in my string - and it would work

belovmd reviewed (commented):
8,5
